server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -287,20 +289,6 @@ static void parseObjectOrNested(DocumentParserContext context) throws IOExceptio
         }
 
         if (context.parent().isNested()) {
-            // Handle a nested object that doesn't contain an array. Arrays are handled in #parseNonDynamicArray.
-            if (context.mappingLookup().isSourceSynthetic() && context.getClonedSource() == false) {
-                Tuple<DocumentParserContext, XContentBuilder> tuple = XContentDataHelper.cloneSubContext(context);
-                context.addIgnoredField(
-                    new IgnoredSourceFieldMapper.NameValue(
-                        context.parent().name(),
-                        context.parent().fullPath().indexOf(context.parent().simpleName()),
-                        XContentDataHelper.encodeXContentBuilder(tuple.v2())
-                    )
-                );
-                context = tuple.v1();
-                token = context.parser().currentToken();
-                parser = context.parser();
-            }
```

Comment:
Let's keep this behind option `store_array_source`.

Reply:
Hm maybe the existing logic for the latter covers it, in case we update `NestedObjectMapper` to set the option properly.